File: homework4/turnIn/helloWorld/helloWorld.f90
PROGRAM helloWorld
INCLUDE 'mpif.h'

INTEGER ierr, rank,nNodes,i

!initialize MPI
CALL MPI_INIT(ierr);
!get number of processors
CALL MPI_COMM_SIZE(MPI_COMM_WORLD, nNodes,ierr)
!get the rank of the processor
CALL MPI_COMM_RANK(MPI_COMM_WORLD,rank,ierr)

PRINT *,"Hello world from processor ", rank, " out of ", nNodes

CALL MPI_Finalize(ierr);

END PROGRAM helloWorld

File: homework4/turnIn/helloWorld/helloWorld.c
#include <stdio.h>
#include <mpi.h>

main(int argc, char **argv){
    int ierr; //the variable that initializes everything
    int nameLen;
    int nNodes;
    int rank;
    
    //initialize MPI
    ierr = MPI_Init(&argc, &argv);
    // get number of processors
    MPI_Comm_size(MPI_COMM_WORLD, &nNodes);
    //get the rank of the processor
    MPI_Comm_rank(MPI_COMM_WORLD,&rank);
    
    printf("Hello world from processor %d out of %d processors\n",rank,nNodes);

    ierr = MPI_Finalize();
 
}
